libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F4/serial_api.c

```diff
@@ -395,13 +531,24 @@ void serial_irq_set(serial_t *obj, SerialIrq irq, uint32_t enable)
 
         if (irq == RxIrq) {
             __HAL_UART_ENABLE_IT(&UartHandle, UART_IT_RXNE);
+#if DEVICE_SERIAL_ASYNCH_DMA
+            NVIC_SetVector(irq_n, vector_dma);
+            NVIC_EnableIRQ(irq_n);
+            NVIC_SetVector(irqn_dma, vector_dma);
+            NVIC_EnableIRQ(irqn_dma);
+#else
+            NVIC_SetVector(irq_n, vector);
+            NVIC_EnableIRQ(irq_n);
+#endif
         } else { // TxIrq
             __HAL_UART_ENABLE_IT(&UartHandle, UART_IT_TC);
+            NVIC_SetVector(irq_n, vector);
+            NVIC_EnableIRQ(irq_n);
+#if DEVICE_SERIAL_ASYNCH_DMA
+            NVIC_SetVector(irqn_dma, vector_dma);
```

Comment:
Isn't this function intended to use only for interrupts, no dma? It is invoked when attach() is called, thus this would enable DMa interrupt for one character?